### DISCUSSION TITLE
Post-Transformed Path from Middleware

### DISCUSSION THREAD

**Discussion by pinkfloydx33** [dotnet/yarp#1513] [Q&A] [answered]
How would you get the proxied/post-transformed path  (not necessarily the full URL) from within custom middleware?

For example, suppose the following configuration:
```json
{
	"Routes": {
		"my-route": {
			"ClusterId": "my-cluster",
			"Match": {
				"Path": "/path/{**catch-all}"
			},
			"Transforms": [
				{
					"PathRemovePrefix": "/path"
				},
				{
					"PathPrefix": "/api"
				}
			]
		}
	},
	"Clusters": {
		"my-cluster": {
			"Destinations": {
				"destination": {
					"Address": "http://my.backend.svc/"
				}
			}
		}
	}
}
```

An incoming path of the form `/path/something` will be translated to `/api/something` after the transforms are applied. With a final destination URL of `http://my.backend.svc/api/something`.  From within middleware, I can inspect the `Request.Path` to get the original path (ie. `/path/something`), but what would I do to get the post-transformed path, `/api/something` (with or without the full destination host):

```csharp
app.UseEndpoints(endpoints => 
{
   endpoints.MapReverseProxy(proxy => 
   {
      proxy.Use(async (ctx, next) => 
      {
         // How to get the transformed path here?
         var tranformedPath = ...;
         await next();
      });
      proxy.UseSessionAffinity();
      proxy.UseLoadBalancing();
      proxy.UsePassiveHealthChecks();
   });
});
```

**Comment by Tratcher** [comment, marked as answer]
Asp.Net middleware don't have access to that, the transformed request path only exists on the outbound HttpRequestMessage. You can capture it in a [custom transform](https://microsoft.github.io/reverse-proxy/articles/transforms.html#extensibility) or in a [DelegatingHandler](https://microsoft.github.io/reverse-proxy/articles/http-client-config.html#custom-iforwarderhttpclientfactory)

**Reply by Tratcher** [reply]
What did you plan on doing with the value?

**Reply by pinkfloydx33** [reply]
I'm not sure what I was thinking at the time. I guess I don't really need it.  I ended up getting what I *did* need with custom endpoint Metadata and didn't need the path at all. 

Though I suppose one might still want to know the actual post-transformed path for logging/debug purposes, especially if using a transform such as `PathPattern` whose results may not be obvious. 

Purely educational at this point--If it were part of a transform, I suppose it would need to be the final one and you'd just grab it from `RequestTransformContext.Path`?

**Reply by Tratcher** [reply]
> Purely educational at this point--If it were part of a transform, I suppose it would need to be the final one and you'd just grab it from `RequestTransformContext.Path`?

Right, though I think YARP already logs it.
https://github.com/microsoft/reverse-proxy/blob/a323abe22131818853ae523420d5aa72bf9d9d93/src/ReverseProxy/Forwarder/HttpForwarder.cs#L302-L304
